### ISSUE TITLE
Remove usage of noble-secp256k1 utils

### ISSUE THREAD

**Issue by paulmillr** [BitGo/BitGoJS#2751]
I have removed them because I think their api is suboptimal. If you want to keep using them, copy-paste this:

https://github.com/paulmillr/noble-secp256k1/blob/37e66841eb155d21b4fe6658223140d5d7d5176e/test/index.ts#L467

**Comment by mmcshinsky-bitgo** [comment]
Thanks for the update! We're making the adjustments currently internally as well as a result of the change.

**Comment by paulmillr** [comment]
As a side note, your api which depended on those functions is also overengineered, e.g. `return bigIntFromU8ABE(secp.utils.privateNegate(s));` could be just `secp.utils.mod(-s, secp.CURVE.n)`. Every type conversion has potential error somewhere, which means something could go wrong.

```
  scalarSub(x: bigint, y: bigint): bigint {
    const negatedY = secp.utils.privateNegate(y);
    return bigIntFromU8ABE(secp.utils.privateAdd(x, negatedY));
  }
  // could just be
    scalarSub(x: bigint, y: bigint): bigint {
    return secp.utils.mod(x - y, secp.CURVE.n)
  }
```

**Comment by mmcshinsky-bitgo** [comment]
That's great to know, thanks!

@brandonblack You may be interested in this.

**Comment by brandonblack** [comment]
Thanks @paulmillr. 

FWIW, removing these in a minor version caused us quite some pain when builds started spontaneously failing due to semver `^`.

**Comment by paulmillr** [comment]
@brandonblack sorry for this, I intend to follow semver, however, I did not even add these methods to docs, and considered them experimental. What should have been done from my end: naming them with `_underscore`. That should have been better communicated from my end.

**Comment by brandonblack** [comment]
Thanks. We'll update to 1.7.0 and expand our wrapper to include these functions.

**Comment by mmcshinsky-bitgo** [comment]
@brandonblack What is the status of this issue? Is this issue now resolved?

**Comment by brandonblack** [comment]
@mmcshinsky-bitgo no. will probably work on it next sprint - this is only an issue when we upgrade noble-secp256k1 to 1.7+
